src/components/Modals/Send/TxFeeRadioGroup.tsx

```diff
@@ -1,84 +1,124 @@
-import { Box, Button, ButtonGroup, Radio, useColorModeValue } from '@chakra-ui/react'
-import { RawText, Text } from 'components/Text'
+import { Box, Button, ButtonGroup, Radio, Spinner, useColorModeValue } from '@chakra-ui/react'
+import { FeeDataKey } from '@shapeshiftoss/chain-adapters'
+import { Amount } from 'components/Amount/Amount'
+import { Text } from 'components/Text'
 import { useController, useFormContext, useWatch } from 'react-hook-form'
 
-// @TODO: Read from state
-const fees = [
-  {
-    name: 'Slow',
-    translation: 'modals.send.sendForm.slow',
-    amount: '24 sat/byte',
-    fiat: '$1.32',
-    color: 'yellow'
-  },
-  {
-    name: 'Average',
-    translation: 'modals.send.sendForm.average',
-    amount: '80 sat/byte',
-    fiat: '$1.49',
-    color: 'blue'
-  },
-  {
-    name: 'Fast',
-    translation: 'modals.send.sendForm.fast',
-    amount: '115 sat/byte',
-    fiat: '$1.76',
-    color: 'green'
+import { FeePrice } from './views/Confirm'
+
+type TxFeeRadioGroupProps = {
+  fees: FeePrice | null
+}
+
+function getFeeColor(key: FeeDataKey): string {
+  switch (key) {
+    case FeeDataKey.Slow:
+      return 'yellow'
+    case FeeDataKey.Fast:
+      return 'green'
+    case FeeDataKey.Average:
+    default:
+      return 'blue'
+  }
+}
+
+function getFeeTranslation(key: FeeDataKey): string {
+  switch (key) {
+    case FeeDataKey.Slow:
+      return 'modals.send.sendForm.slow'
+    case FeeDataKey.Fast:
+      return 'modals.send.sendForm.fast'
+    case FeeDataKey.Average:
+    default:
+      return 'modals.send.sendForm.average'
   }
-]
+}
+
+const feesOrder: FeeDataKey[] = [FeeDataKey.Slow, FeeDataKey.Average, FeeDataKey.Fast]
 
-export const TxFeeRadioGroup = () => {
+export const TxFeeRadioGroup = ({ fees }: TxFeeRadioGroupProps) => {
   const { control } = useFormContext()
   const { field } = useController({
-    name: 'fee',
+    name: 'feeType',
     control,
     rules: { required: true },
-    defaultValue: 'Average'
+    defaultValue: 'average'
```

Comment:
what is "useController"? Why are these strings? Should there be an ENUM to define the possible values (or type them as a string union to avoid typos).

Reply:
useController is a hook from `react-hook-form` used to turn uncontrolled inputs into controlled inputs. I like the idea of using an enum to define the required form fields to ensure that people are using the right field name.